TACC/tup-ui
# feat(tup-cms): redirect plugin app

## Status

Failed informal review by designer and peer developer. See #128 instead.

## Overview

Allow news articles to external news service without learning and changing the plugin.

<details>The news plugin does not support external links, so this was an idea to give us support that without dissecting the plugin.</details>

Another use case (internal) would be [the redirect to automatic pattern library](https://dev.tup.tacc.utexas.edu/design-system/pattern-library/).

## Related

- replaced by #128.

## Changes

- add `tacc_redirect` plugin

## Testing

In each of these different contexts:
- **A.** be logged **in** and in **edit** mode with javascript **on**
- **B.** do **not** be in **edit** mode (logged or not)
- **C.** be logged **in** and in **edit** mode have javascript **off**

Perform these steps:
1. Open https://dev.tup.tacc.utexas.edu/news/latest-news/.
2. Click [Towers in the Storm](https://dev.tup.tacc.utexas.edu/news/latest-news/2022/12/22/towers-storm/) article.
3. Verify:
    - **context A.** Message about external article (with link). Message about redirect (with link). Message replaced by message cancelling redirect. No redirect.
    - **context B.** Message about external article (with link). Message about redirect (with link). Redirect occurs.
    - **context C.** Message about external article (with link).

## UI

Skipped.
